deepchecks/utils/abstracts/confusion_matrix_abstract.py

```diff
@@ -77,3 +79,77 @@ def create_confusion_matrix_figure(confusion_matrix_data: np.ndarray, classes_na
+    """Condition function that checks if the misclassified samples in the confusion matrix is below threshold.
+
+    Parameters
+    ----------
+    value: np.ndarray
+        Value of the confusion matrix
+    misclassified_samples_threshold: float
+        Ratio of samples to be used for comparison in the condition (Value should be between 0 - 1 inclusive)
+
+    Raises
+    ------
+    ValidationError
+        if the value of `misclassified_samples_threshold` parameter is not between 0 - 1 inclusive.
+
+    Returns
+    -------
+    ConditionResult
+        - ConditionCategory.PASS, if all the misclassified samples in the confusion
+        matrix are less than `misclassified_samples_threshold` ratio
+        - ConditionCategory.FAIL, if the misclassified samples in the confusion matrix
+        are more than the `misclassified_samples_threshold` ratio
+    """
+    if misclassified_samples_threshold < 0 or misclassified_samples_threshold > 1:
+        raise ValidationError(
+           'Condition requires the parameter "misclassified_samples_threshold" '
+           f'to be between 0 and 1 inclusive but got {misclassified_samples_threshold}'
+        )
+
+    # Computing the total number of samples from the confusion matrix
+    total_samples = np.sum(value)
+
+    # Number of threshold samples based on the 'misclassified_samples_threshold' parameter
+    thresh_samples = round(np.ceil(misclassified_samples_threshold * total_samples))
+
+    # m is the number of rows in the confusion matrix and
+    # n is the number of columns in the confusion matrix
+    m, n = value.shape[0], value.shape[1]
+
+    # Variables to keep track of cells above 'thresh_samples'
+    n_cells_above_thresh = 0
+    max_samples_in_cell_above_thresh = thresh_samples
+
+    # Looping over the confusion matrix and checking only the misclassified cells
+    for i in range(m):
+        for j in range(n):
+            # omitting the principal axis of the confusion matrix
+            if i != j:
+                n_samples = value[i][j]
+
+                if n_samples > thresh_samples:
+                    n_cells_above_thresh += 1
+
+                    if n_samples > max_samples_in_cell_above_thresh:
+                        max_samples_in_cell_above_thresh = n_samples
+
+    # There are misclassified cells in the confusion matrix with samples more than 'thresh_samples'
+    if n_cells_above_thresh > 0:
+        details = f'The confusion matrix has {n_cells_above_thresh} cells with samples ' \
+                  f'greater than the threshold ({thresh_samples}) based on the ' \
+                  'given misclassified_samples_threshold ratio. The worst performing cell ' \
+                  f'has {max_samples_in_cell_above_thresh} samples'
+
+        return ConditionResult(ConditionCategory.FAIL, details)
+
+    # No cell has more than 'thresh_samples' misclassified samples
+    details = 'Number of samples in each of the misclassified cells in the ' \
```

Comment:
change to:
All misclassified matrix cells contain less then {} percent of the data

Reply:
Done.